src/Symfony/Component/JsonPath/JsonPathUtils.php

```diff
@@ -263,4 +263,52 @@ public static function isIntegerOverflow(string $s): bool
 
         return $negative ? $s < '-9007199254740991' : $s > '9007199254740991';
     }
+
+    /**
+     * Transforms JSONPath regex patterns to comply with RFC 9535.
+     *
+     * The main issue is that '.' should not match \r or \n but should
+     * match Unicode line separators U+2028 and U+2029.
+     */
+    public static function normalizeRegex(string $pattern): string
+    {
+        $result = '';
+        $inCharClass = false;
+        $escaped = false;
+        $i = -1;
+
+        while (null !== $char = $pattern[++$i] ?? null) {
+            if ($escaped) {
+                $result .= $char;
+                $escaped = false;
+                continue;
+            }
+
+            if ('\\' === $char) {
+                $result .= $char;
+                $escaped = true;
+                continue;
+            }
+
+            if ('[' === $char && !$inCharClass) {
+                $inCharClass = true;
+                $result .= $char;
+                continue;
+            }
+
+            if (']' === $char && $inCharClass) {
+                $inCharClass = false;
+                $result .= $char;
+                continue;
+            }
+
+            if ('.' === $char && !$inCharClass) {
+                $result .= '(?:[^\r\n]|\x{2028}|\x{2029})';
+            } else {
+                $result .= $char;
+            }
```

Comment:
there's no need for handling U+2028 / U+2029 specially:

```suggestion
        $i = -1;

        while (null !== $char = $pattern[++$i] ?? null) {
            switch ($char) {
                case '\\': $char .= $pattern[++$i] ?? ''; break;
                case '[': $inCharClass = true; break;
                case ']': $inCharClass = false; break;
                case '.': $inCharClass || $char = '[^\r\n]'; break;
            };

            $result .= $char;
```